pkg/kubefed2/join.go

```diff
@@ -374,13 +378,17 @@ func createFederatedCluster(fedClientset *fedclient.Clientset, joiningClusterNam
 		return fedCluster, nil
 	}
 
-	return fedClientset.CoreV1alpha1().FederatedClusters(federationNamespace).Create(fedCluster)
+	fedCluster, err := fedClientset.CoreV1alpha1().FederatedClusters(federationNamespace).Create(fedCluster)
+	if idempotent && errors.IsAlreadyExists(err) {
```

Comment:
How about create a function named as `IsIdempotentMode` to replace `idempotent && errors.IsAlreadyExists(err)`?

/cc @xunpan

Reply:
Something like
```go
if ignoreIfIdempotent(idempotent, err) {
...
}
```
?
imho, I would think that it would obscure the fact that we're only ignoring the error when the item already exists. But I can make the change if you feel strongly about it.